interfaces/repo.go

```diff
@@ -576,19 +570,54 @@ func (r *Repository) Connect(ref *ConnRef, plugDynamicAttrs, slotDynamicAttrs ma
 	// Ensure that such plug exists
 	plug := r.plugs[plugSnapName][plugName]
 	if plug == nil {
-		return nil, &UnknownPlugSlotError{Msg: fmt.Sprintf("cannot connect plug %q from snap %q: no such plug", plugName, plugSnapName)}
+		return nil, nil, &UnknownPlugSlotError{Msg: fmt.Sprintf("cannot connect plug %q from snap %q: no such plug", plugName, plugSnapName)}
 	}
 	// Ensure that such slot exists
 	slot := r.slots[slotSnapName][slotName]
 	if slot == nil {
-		return nil, &UnknownPlugSlotError{fmt.Sprintf("cannot connect slot %q from snap %q: no such slot", slotName, slotSnapName)}
+		return nil, nil, &UnknownPlugSlotError{fmt.Sprintf("cannot connect slot %q from snap %q: no such slot", slotName, slotSnapName)}
 	}
 	// Ensure that plug and slot are compatible
 	if slot.Interface != plug.Interface {
-		return nil, fmt.Errorf(`cannot connect plug "%s:%s" (interface %q) to "%s:%s" (interface %q)`,
+		return nil, nil, fmt.Errorf(`cannot connect plug "%s:%s" (interface %q) to "%s:%s" (interface %q)`,
 			plugSnapName, plugName, plug.Interface, slotSnapName, slotName, slot.Interface)
 	}
 
+	return plug, slot, nil
+}
+
+// Connect re-establishes a connection between a plug and a slot as part of connection reloading.
+// When connections are reloaded policyCheck is null (we don't check policy again).
+func (r *Repository) ConnectReload(ref *ConnRef, plugStaticAttrs, plugDynamicAttrs, slotStaticAttrs, slotDynamicAttrs map[string]interface{}) (*Connection, error) {
```

Comment:
I'm missing part of the picture for these API changes. What does it mean to connect-reload? Is it connecting, or is reloading? What does it mean to reload a connection, and why do we fiddle with its static attributes every time we do that?

Reply:
Reloading means restoring connections in the repository from "conns" data in the state. We do that in the ifacemgr startup code (InterfaceManager - `reloadConnections` method). In the existing code we just call `repo.Connect(...)` there, but the problem is that it re-creates a connection with static attributes coming from snap yaml data, and dynamic attributes coming from whatever was stored in `conns`. The `ConnectReload` method here allows for re-creating `ConnectedPlug|Slot` which both static and dynamic attributes coming from `conns`.
I agree, the name is probably not very fortunate.